docs/model_access_control.md

```diff
@@ -0,0 +1,652 @@
+# Model Access Control
+
+In the present setting, on a security enabled OpenSearch cluster, we can limit users to certain actions by mapping them to relevant permissions and permission roles. For example, we may want some users to only search and view ml models while allow others to register and deploy them. 
+
+Currently, ml-commons has two pre-defined roles to manage permissions. For descriptions of each, see [Permissions](https://opensearch.org/docs/latest/ml-commons-plugin/index/#permissions).
+Users can also mix and match individual ml permissions to suit their use case, if these roles don’t meet their needs. Each action corresponds to an operation in the REST API. For example, the `cluster:admin/opensearch/ml/models/delete` permission lets the user delete models.
+
+So far, the ml-commons plugin did not have the concept of ownership. For example, if a user has the `cluster:admin/opensearch/ml/models/deploy` permission, they can deploy all models, regardless of whether who registered them. So, we aim to build a security-based access control so that only certain users have access to models created/owned by other users. That means, even if users have delete model permission, they can delete a resource only if they have access to it which is determied by the access mode of that resource. This limits the resources a user can have access to on the cluster therby reducing the security risks. 
+
+We have implemented backend role-based access in which members of the same back-end roles will be able to access each other’s resources. This is similar to alerting and anomaly detection plugin [backend role based access](https://opensearch.org/docs/latest/observing-your-data/alerting/security/#advanced-limit-access-by-backend-role). Additionally we also let users define an ml resource as public or private. This advanced security access is available only on a cluster with [Security plugin](https://opensearch.org/docs/latest/security/index/).
+
+Note: 
+- Model access control is experimental feature. If you see any bug or have any suggestion, feel free to cut Github issue.
+
+
+
+## Setup
+
+For Backend Role based security, it is not enabled by default. To enable/disable it, customers can toggle elasticsearch cluster setting shown below. If this setting is not enabled, a resource can be accessed by anyone.
```

Comment:
Just curious, to create a model group, do we need to enable this setting?

Reply:
not necessary for creating model group. This setting is only to enable access control

Reply:
hmm, I just tried to create a model group:

```
POST /_plugins/_ml/model_groups/_register
{
    "name": "test_model_group_public",
    "description": "This is a public model group",
    "model_access_mode": "public"
}
```

And received error :

```
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "Cluster security plugin not enabled or model access control no enabled, can't pass access control data in request body"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "Cluster security plugin not enabled or model access control no enabled, can't pass access control data in request body"
  },
  "status": 400
}
```

But after enabling the `plugins.ml_commons.model_access_control_enabled` then I was able to create the model group:

```
{
  "model_group_id": "rJTkoYgBU4BPBnjEiV4f",
  "status": "CREATED"
}
```

Reply:
So here, in this case, all model groups are public by default. We do not want user to specify the model_access_mode when the setting/security is disabled. Try this without the setting enabled. It would work